nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -190,12 +223,53 @@ in {
         type = lib.types.port;
         default = 55002;
       };
+
       sidekiqThreads = lib.mkOption {
-        description = lib.mdDoc "Worker threads used by the mastodon-sidekiq service.";
+        description = lib.mdDoc "Worker threads used by the mastodon-sidekiq-all service. If `sidekiqProcesses` is configured and any processes specify null `threads`, this value is used.";
         type = lib.types.int;
         default = 25;
       };
 
+      sidekiqProcesses = lib.mkOption {
+        description = lib.mdDoc "How many Sidekiq processes should be used to handle background jobs, and which job classes they handle.";
+        type = with lib.types; attrsOf (submodule {
+          options = {
+            jobClasses = lib.mkOption {
+              type = listOf (enum [ "default" "push" "pull" "mailers" "scheduler" "ingress" ]);
+              description = lib.mdDoc "If not empty, which job classes should be executed by this process. *Only one process should handle the 'scheduler' class.*";
+            };
+            threads = lib.mkOption {
+              type = nullOr int;
+              description = lib.mdDoc "Number of threads this process should use for executing jobs. If null, the configured `sidekiqThreads` are used.";
+            };
+          };
+        });
+        default = {
+          all = {
+            jobClasses = [ ];
+            threads = null;
+          };
+        };
+        example = {
+          all = {
```

Comment:
Is `all` a special sidekiq queue? Couldn't find documentation on it on the fly.

Reply:
`all` is a label I made up because the way I structured this, I needed to label the default *something* and thought this was reasonable. I'm open to suggestions and possibly the complexity of figuring out how to create an un-labelled unit by default, if you want

Reply:
I think it is a good name on our side, but I'm wondering how sidekiq would create the correct queues when being run with the default configuration? It would be started with `-q all` then, right?

Reply:
yep, exactly 👍